homeassistant/components/noonlight/switch.py

```diff
@@ -0,0 +1,118 @@
+"""Create a switch to trigger an alarm in Noonlight"""
+import logging
+
+from datetime import timedelta
+
+from homeassistant.components import persistent_notification
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.helpers.event import async_track_time_interval
+
+from . import (DOMAIN, EVENT_NOONLIGHT_TOKEN_REFRESHED,
+               NOTIFICATION_ALARM_CREATE_FAILURE)
+
+DEFAULT_NAME = 'Noonlight Switch'
+
+CONST_ALARM_STATUS_ACTIVE = 'ACTIVE'
+CONST_ALARM_STATUS_CANCELED = 'CANCELED'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Create a switch to create an alarm with the Noonlight service"""
+    noonlight_platform = hass.data[DOMAIN]
+    noonlight_switch = NoonlightSwitch(noonlight_platform)
+    async_add_entities([noonlight_switch])
+
+    def noonlight_token_refreshed(event):
+        noonlight_switch.schedule_update_ha_state()
+
+    hass.bus.async_listen(EVENT_NOONLIGHT_TOKEN_REFRESHED,
```

Comment:
Don't use the event bus for intra package communication. We have our dispatch helper for this purpose.

Here's an example from heos:
https://github.com/home-assistant/home-assistant/blob/693fa1592485073d60c3e3de9f254ca8465177aa/homeassistant/components/heos/media_player.py#L94-L107

https://github.com/home-assistant/home-assistant/blob/693fa1592485073d60c3e3de9f254ca8465177aa/homeassistant/components/heos/__init__.py#L158-L164